### DISCUSSION TITLE
make all run platform=board error

### DISCUSSION THREAD

**Discussion by Brittanyzhou** [bitcraze/discussions#1165] [Q&A] [answered]
I connect Jtag to flash to the crazyfile, but error received like this:
For bug reports, read
        http://openocd.org/doc/doxygen/bugs.html
DEPRECATED! use 'adapter driver' not 'interface'
DEPRECATED! use 'adapter speed' not 'adapter_khz'
DEPRECATED! use 'adapter driver' not 'interface'
Warn : Interface already configured, ignoring
TARGET create
Info : core 0 selected
Info : gap8_adv_debug_itf tap selected
Info : adv_dbg_unit debug unit selected
Info : Option 7 is passed to adv_dbg_unit debug unit
GAP8 INIT TARGET
Error: no device found
Error: unable to open ftdi device with vid 0403, pid 6010, description '*', serial '*' at bus location '*'

Traceback (most recent call last):
  File "/home/tlabstaff/loc_test/gap_sdk/tools/gapy/gapy", line 162, in <module>
    main()
  File "/home/tlabstaff/loc_test/gap_sdk/tools/gapy/gapy", line 152, in main
    operationFunc(args, config)
  File "/home/tlabstaff/loc_test/gap_sdk/tools/gapy/run.py", line 146, in operationFunc
    raise RuntimeError('Runner has failed with value: %d' % status)
RuntimeError: Runner has failed with value: -1
make: *** [/home/tlabstaff/loc_test/gap_sdk/tools/rules/pulp_rules.mk:232: flash_fs] Error 1.

I try to append the line "UBSYSTEM=="usb", ATTRS{idVendor}=="0403", ATTRS{idProduct}=="6010", MODE="0664", GROUP="plugdev"" to the 99-bitcraze.rules, still same error.

Besides, it runs successfully if the command is "make all run platform=gsvoc".

Please Help!!!

**Comment by knmcguire** [comment]
Hi! It seems to not find your device, so probably you are not using a programmer that it is suspecting.

We need more information about your system:
- Which OS do you run on the computer?
- Which programmer are you using?
- Which documentation/guide are you following?

**Reply by Brittanyzhou** [reply]
- I use ubuntu 22.04
- crazyfile 2.1
- https://github.com/GreenWaves-Technologies/gap_sdk
Seems I solve the problem,but I'm not sure. Here is the new error state:
 warn : Burst read timed out
Error: Burst read failed
can't read halt state
Warn : Debug IF CPU control reg read failure.
Warn : Resetting JTAG TAP state and reconnectiong to debug IF.
Init jtag
Initialising GAP8 JTAG TAP
Info : adv debug unit is configured with option BOOT MODE JTAG
Info : adv debug unit is configured with option ADBG_USE_HISPEED
Warn : Burst read timed out
Warn : Burst read timed out
Warn : Burst read timed out
Warn : Burst read timed out

**Reply by knmcguire** [reply]
Oke, for question 2, I meant, which JTAG programmer do you use. is this an Olimex, stlink or j-link programmer

for question 3. it seems that you are not flashing the crazyflie but the AIdeck that is attached to it, because the crazyflie firmware does not rely on the gap_sdk. Have you found the aideck getting started guide yet? https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/

**Reply by Brittanyzhou** [reply]
I used Olimex 

So I need to flash wifi before I deploy the application to AI deck, right?

**Reply by knmcguire** [reply]
Well you do need to go through the steps of the tutorial first. Once that is done successfully then you can move on. The wifi example is just the easiest to use to check if everything is working on your end.

**Reply by Brittanyzhou** [reply]
I tried. The reason is the AI deck doesn't connect with Olimex correctly.  The AI deck may be some problems.

**Reply by gemenerik** [reply]
Can you be a little more specific?  Do I correctly deduce that you can run the [getting started guide](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/) up to the point where you have to [flash the GAP8 bootloader](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#gap8-bootloader)?

Are you using the docker to flash the GAP8?
Have you connected your JTAG to the GAP8 correctly? Is it not connected to the ESP ("NINA JTAG")? Is it the correct way around?
Is your deck correctly powered when you try to flash it?

**Reply by Brittanyzhou** [reply]
Yes, I can run the getting started guide after flash the GAP8 bootloader. 

Yes, I used the docker to flash the GAP8.

Yes, I connected JTAG to the GAP8 correctly. 

Yes, my deck correctly powered when I try to flash it. 

The error occured may be loose or defective contact between Jtag and AI deck.

**Reply by knmcguire** [reply]
I'm a bit worried about this error from your terminal output since this doesn't seem to be the same as Olimex programmer but I'm not exactly sure which it is supposed to be. 

`Error: unable to open ftdi device with vid 0403, pid 6010, description '', serial '' at bus location '*'
`

Could you give the full command you enter in the terminal to startup the gap8 docker and flash (like `docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 ...`

Also, could you give the output when you enter `lsusb` in the terminal?

**Reply by Brittanyzhou** [reply, marked as answer]
Hi, thanks for your help. I have already solved this problem and the error appeared is olimx connection error.

**Reply by knmcguire** [reply]
alright thanks for letting us know. Next time its better to leave the discussion open and select an answer (or write the fix you have done yourseslf)
